components/firebase_admin_sdk/firebase_admin_sdk.app.mjs

```diff
@@ -76,52 +70,44 @@ module.exports = {
         data,
         params,
       };
-      return (await axios(config)).data;
+      return axios(this, config);
     },
     /**
      * Retrieves a Bearer token for use with the Firebase REST API.
-     * @param {string} apiKey - the Web API Key, which is obtained from the project
-     * settings page in the admin console
      * @returns {object} returns an object containing a new token and refresh token
      */
-    async _getToken(apiKey) {
-      const { clientEmail } = this.$auth;
-      const newCustomToken = await admin
-        .auth()
-        .createCustomToken(clientEmail)
-        .catch((error) => {
-          console.log("Error creating custom token:", error);
-        });
-      const data = {
-        token: newCustomToken,
-        returnSecureToken: true,
-      };
-      const params = {
-        key: apiKey,
-      };
-      return await this._makeRequest(
-        "POST",
-        "https://identitytoolkit.googleapis.com/v1/accounts:signInWithCustomToken",
-        data,
-        params,
+    async _getToken() {
+      const {
+        clientEmail,
+        privateKey,
+      } = this.$auth;
+      const formattedPrivateKey = privateKey.replace(/\\n/g, "\n");
+      const SCOPES = [
+        "https://www.googleapis.com/auth/cloud-platform",
+      ];
+      const jwtClient = await new googleAuth.JWT(
+        clientEmail,
+        null,
+        formattedPrivateKey,
+        SCOPES,
+        null,
       );
+      return jwtClient.authorize().then((tokens) => tokens.access_token);
```

Comment:
Since we are using `async/await` pattern aproach I recommend using it here to return the token
```suggestion
      const { access_token: accessToken } = await jwtClient.authorize();
      return accessToken;
```